pkg/koordlet/metricsadvisor/collector.go

```diff
@@ -263,33 +276,36 @@ func (c *collector) collectPodResUsed() {
 			klog.Errorf("insert pod %s/%s, uid %s resource metric failed, metric %v, err %v",
 				pod.Namespace, pod.Name, uid, podMetric, err)
 		}
-		c.collectContainerResUsed(meta)
+
 	}
 
 	// update collect time
 	c.state.RefreshTime(podResUsedUpdateTime)
 	klog.Infof("collectPodResUsed finished, pod num %d", len(podMetas))
 }
 
-func (c *collector) collectContainerResUsed(meta *statesinformer.PodMeta) {
+func (c *collector) collectContainerResUsed(meta *statesinformer.PodMeta) (totalGPUUsages []*metriccache.GPUMetric) {
 	klog.V(6).Infof("start collectContainerResUsed")
 	pod := meta.Pod
+	metricMap := make(map[int32]*metriccache.GPUMetric, 0)
 	for i := range pod.Status.ContainerStatuses {
 		containerStat := &pod.Status.ContainerStatuses[i]
 		collectTime := time.Now()
 		currentCPUUsage, err0 := util.GetContainerCPUUsageNanoseconds(meta.CgroupDir, containerStat)
 		memUsageValue, err1 := util.GetContainerMemStatUsageBytes(meta.CgroupDir, containerStat)
-		if err0 != nil || err1 != nil {
+		currentPID, err2 := util.GetContainerPID(meta.CgroupDir, containerStat)
+		if err0 != nil || err1 != nil || err2 != nil {
 			// higher verbosity for probably non-running pods
 			if containerStat.State.Running == nil {
 				klog.V(6).Infof("failed to collect non-running container usage for %s/%s/%s, "+
-					"CPU err: %s, Memory err: %s", pod.Namespace, pod.Name, containerStat.Name, err0, err1)
+					"CPU err: %s, Memory err: %s, PID err: %s", pod.Namespace, pod.Name, containerStat.Name, err0, err1, err2)
 			} else {
 				klog.Warningf("failed to collect container usage for %s/%s/%s, CPU err: %s, Memory err: %s",
 					pod.Namespace, pod.Name, containerStat.Name, err0, err1)
```

Comment:
GetContainerPID error is needed here as well

Reply:
done